src/lib/isAfter.js

```diff
@@ -1,9 +1,11 @@
-import assertString from './util/assertString';
 import toDate from './toDate';
 
-export default function isAfter(str, date = String(new Date())) {
-  assertString(str);
-  const comparison = toDate(date);
-  const original = toDate(str);
+export default function isAfter(date, options) {
+  // For backwards compatibility:
+  // isAfter(str [, date]), i.e. `options` could be used as argument for the legacy `date`
+  const comparisonDate = options?.comparisonDate || options || Date().toString();
```

Comment:
```suggestion
export default function isAfter(date, ...args) {
  // For backwards compatibility:
  // isAfter(str [, date]), i.e. `options` could be used as argument for the legacy `date`
  const comparisonDate = (typeof args[0] === 'object' ? args[0].comparisonDate : args[0]) || Date().toString();
```

@pixelbucket-dev what do you think about this? Not necessarily the nicer solution, but would be more in line with what you've done for `isAlpha(numeric)`

Reply:
I don't think you have to be consistent here. This is a feature to maintain backwards compatibility for the legacy interface, which will soon be removed anyway. The closer you get the code to the future (clean) interface, the better, because it will be easier to remove the legacy args handling in a future PR. So your original solution is not just closer to the future version with only the options arg, it is also less code and easier to read.

The reason my solution is more complex is, that `isAlpha` and `isAlphanumeric` already had an argument before the already exiting `options`argument. This made it harder to find a clean backwards-compatible implementation. Otherwise, I would have just put one `options` arg in the function signature 😄.

However, what we could think about is to create a shared function for `isBefore` and `isAfter`, since they share most code. Similar to the `validate` function, I've created for `isAlpha` and `isAlphanumeric`. The strategy here is composition. The reason for it: These PRs revealed that it is hard to synch between the PRs when you have two functions with almost the same code, like your `isAfter` and my `isBefore`. It would have been much easier to change only the shared behaviour in one PR, i.e. the generic implementation. The specialised `isBefore` and `isAfter` could have been untouched. I can do a follow-up PR after these two to create a shared function. I am not yet sure where to put it. TBH I am not happy that I've put `validate`, `isAlpha` and `isAlphanumeric` in the `alpha.js` file. There should probably be a separate file inside the util folders. 

Which brings me to another discussion: We, rightfully, change the file suffix for test files to `.test.js`. In my own experience, it is a good practice to do that for all specialised files. E.g. we should also do it for the util files in `.util.js`. WDYT?

Reply:
What I'm not unsure of is when we will remove the legacy interface. But I agree that the solution I have now is nicer so I'll keep it like this. Feel free to work on a follow-up PR for a shared function.

I'm not that familiar with different file suffixes than for test but I'm open to the idea. Although I think it would also be fine if you just move the new functions of `alpha.js` file to the util folder

Reply:
I agree about naming stuff `.test.js`! I've not used the `.util.js` naming convention before, but I like it!

Reply:
Let's do `.util.js` in a future PR